File: DSRFQ.Web/Modules/ServerTypes/Master/CurrenciesRow.ts
import { fieldsProxy } from '@serenity-is/corelib';

export interface CurrenciesRow {
    Id?: number;
    Code?: string;
    Name?: string;
    InsertDate?: string;
    InsertUserId?: number;
    UpdateDate?: string;
    UpdateUserId?: number;
    DeleteDate?: string;
    DeleteUserId?: number;
    IsActive?: number;
}

export abstract class CurrenciesRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Code';
    static readonly localTextPrefix = 'Master.Currencies';

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<CurrenciesRow>();
}

File: DSRFQ.Web/Modules/Master/Currencies/CurrenciesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;
using DSRFQ.Modules.Common.Permissions;

namespace DSRFQ.Master;

[ConnectionKey("Default"), Module("Master"), TableName("MasterCurrencies")]
[DisplayName("Currencies"), InstanceName("Currencies")]
[NavigationPermission(MasterPermissionKeys.MasterCurrencyNavigation)]
[ReadPermission(MasterPermissionKeys.MasterCurrencyView)]
[DeletePermission(MasterPermissionKeys.MasterCurrencyDelete)]
[UpdatePermission(MasterPermissionKeys.MasterCurrencyUpdate)]
[InsertPermission(MasterPermissionKeys.MasterCurrencyInsert)]
[ServiceLookupPermission("?")]
[LookupScript("MasterCurrencies",Permission = "?")]
public sealed class CurrenciesRow : LoggingRow<CurrenciesRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Column("ID"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Code"), Size(50), NotNull, QuickSearch, NameProperty]
    public string Code { get => fields.Code[this]; set => fields.Code[this] = value; }

    [DisplayName("Name"), Size(100), NotNull]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }


    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public StringField Code;
        public StringField Name;
       

    }
}